featuretools/feature_base/feature_base.py

```diff
@@ -141,35 +149,46 @@ def get_dependencies(self, deep=False, ignored=None, copy=True):
 
 
         """
-        deps = []
+        hash_key = hash(f"{self.get_name()}{self.dataframe_name}{deep}{ignored}")
+        if cached_dependencies := self.cache.get(CacheType.DEPENDENCY, hash_key):
+            return cached_dependencies
 
-        for d in self.base_features[:]:
-            deps += [d]
+        ignored = ignored if ignored else set()
```

Comment:
I think you could also do `ignored = ignored or set()` here if you wanted.

Reply:
Good call!